### ISSUE TITLE
"&amp;" html entity in the name of the testcase became "&" if is followed by a ";" not preceded by a "&" (xmlbuilder2 issue)

### ISSUE THREAD

**Issue by Laurianti** [bhovhannes/junit-report-merger#127]
Hello, this is a strange case:

![image](https://user-images.githubusercontent.com/20415513/136624759-9f867668-6864-4044-a9eb-7375e67e3930.png)

This is the combined output:

![image](https://user-images.githubusercontent.com/20415513/136624801-867dc5a6-ed9f-47d2-b90e-c282dd0fed68.png)

I think this is an issue of the xmlbuilder2 library.
https://github.com/oozcitak/xmlbuilder2/issues/109

**Comment by ggrossetie** [comment]
I can reproduce this: https://github.com/michaelleeallen/mocha-junit-reporter/issues/181

**Comment by ggrossetie** [comment]
@bhovhannes we've created a fork to fix this issue: https://www.npmjs.com/package/@datagalaxy/xmlbuilder2

Here's the full diff: 

https://github.com/oozcitak/xmlbuilder2/compare/master...datagalaxy-lab:xmlbuilder2:master

The minimum Node version is now 14 and we've also upgraded the dependencies to the latest version.
Ozgur Ozcitak (`oozcitak`) is not really active anymore on GitHub so I'm not sure if `xmlbuilder2` will receive any update.

Let me know if you are willing to switch to our fork 😉

**Comment by bhovhannes** [comment]
I added a failing test case for this issue and downgraded `xmlbuilder2` dependency to `3.0.1`. With `3.0.1` test passed.
It looks like from `3.0.1` to `3.0.2` Ozgur closed https://github.com/oozcitak/xmlbuilder2/issues/99 and comments under that issue are not convincing that it was indeed fixed.

@ggrossetie I'll keep an eye on your fork and will consider switching to it once it will get more traction. I think fork will get more traction if 

- you refer to it in https://github.com/oozcitak/xmlbuilder2/issues/109 and https://github.com/oozcitak/xmlbuilder2/issues/110. There are number of other encoding issues in xmlbuilder2 and maybe some of them will be fixed by your commit in your fork as well. When people will notice that there is a fork which tries to fix something they will try if the fork works in their use case.
- update fork Readme to mention that it is the fork and describe the reasoning behind creating a fork.

**Comment by universalhandle** [comment]
Greetings! I arrived here from a link in the official `xmlbuilder2` repo. I've just joined that project as a new maintainer. From a quick read I think the issues raised here are fixed in https://github.com/oozcitak/xmlbuilder2/pull/154.

I'm still working on getting maintainer status in npm; as soon as I have that I'll ship the fix in question. I hope we can coalesce around the official package rather than end up with various forks -- I'm very open to collaboration. Cheers!
